### PR TITLE
Deepcopy fix

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

File: codegen/model/resource.go [hunk 1]
```diff
@@ -60,6 +61,19 @@ type Resource struct {
...
+	AdditionalPrinterColumns []apiextv1beta1.CustomResourceColumnDefinition
+}
+
+type ExternalPackage struct {
+	// The go package where the status and spec field can be found, if different than group root
+	// leave empty if same as output dir
+	GoPackage string
+	// The relative path to the package where the spec and status field can be found
+	// leave empty if same as output dir
+	RelativePath string
```
Comment (ilackarms) [inline codegen/model/resource.go]: let's discuss options for removing this

Comment (EItanya) [reply codegen/model/resource.go]: Oh yea, I forgot about that, I will fix later

File: codegen/render/funcs.go [hunk 2]
```diff
@@ -51,15 +52,24 @@ func makeTemplateFuncs() template.FuncMap {
...
+			var result []string
+
 			for _, file := range uniqueFile {
 				for _, desc := range file.GetMessageType() {
-					// for each message, in each file, find the fields which need deep copy functions
-					result = append(result,
-						findDeepCopyFields(file.GetPackage(), grp.Resources, desc, grp.goPackageToMatch != "")...
-					)
+					switch {
+					case grp.rootGoPackage == file.GetOptions().GetGoPackage():
```
Comment (ilackarms) [inline codegen/render/funcs.go:64]: do you mind commenting the cases in this switch?

Comment (ilackarms) [reply codegen/render/funcs.go:64]: as in - why we want to deepcopy on that message

File: codegen/render/funcs.go [hunk 3]
```diff
@@ -48,6 +86,42 @@ func makeTemplateFuncs() template.FuncMap {
 	return f
 }
 
+
+
+/*
+	Find the proto messages for a given set of descriptors which need proto_deepcopoy funcs and whose types are not in
+	the API root package
```
Comment (ilackarms) [inline codegen/render/funcs.go:93]: comment: return true if spec or status + in an external package

**Review by ilackarms — APPROVED** [review]

**Pull request merged**